apiary.apib

```diff
@@ -398,11 +398,318 @@ MerkleDAG.
 
 # Group bitswap
 
-## stat
+A set of commands to manipulate the bitswap agent
+
+## stat [GET /bitswap/stat]
+Show some diagnostic information on the bitswap agent
+
++ Request
+
+    #### curl
+
+        curl -i http://localhost:5001/api/v0/bitswap/stat
+
+    + Body
+
+        ```
+        curl -i http://localhost:5001/api/v0/bitswap/stat
+        ```
+
++ Response 200 (application/json)
+
+    + Attributes (object)
+        + ProvideBufLen: 0 (number)
+        + Wantlist (array, nullable)
+        + Peers (array)
+            - QmNjRVohhWBX31EoaAXkrj5mPF9vQNcTVvQgWHNwdxweCN (Multihash)
+        + BlocksReceived: 0 (number)
+        + DupBlksReceived: 0 (number)
+        + DupDataReceived: 0 (number)
+
+    + Headers
+
+        ```
+        Access-Control-Allow-Headers: X-Stream-Ouptut, X-Chunked-Output
+        Access-Control-Expose-Headers: X-Stream-Ouptut, X-Chunked-Output
+        Trailer: X-Stream-Error
+        Transfer-Encoding: chunked
+        Date: Thu, 21 Jan 2016 17:09:00 GMT
+        Transfer-Encoding: chunked
+        ```
+
+    + Body
 
-## unwant
+        ```
+        {
+          "ProvideBufLen": 0,
+          "Wantlist": null,
+          "Peers": [
+            "QmSoLPppuBtQSGwKDZT2M73ULpjvfd3aZ6ha4oFGL1KrGM",
+            "QmSoLnSGccFuZQJzRadHn95W2CrSFmZuTdDWP8HXaHca9z",
+            "QmSoLueR4xBeUbY9WZ9xGUUxunbKWcrNFTDAadQJmocnWm"
+          ],
+          "BlocksReceived": 0,
+          "DupBlksReceived": 0,
+          "DupDataReceived": 0
+        }
+        ```
+
+## unwant [GET /bitswap/unwant{?arg}]
+Remove a given block from your wantlist
+
++ Parameters
+    + arg (Multihash, required) - key to remove from your wantlist
+
++ Request No Arguments
+
+    #### curl
+
+        curl -i http://localhost:5001/api/v0/bitswap/unwant
+
+    + Body
+
+        ```
+        curl -i http://localhost:5001/api/v0/bitswap/unwant
+        ```
+
++ Response 400
+
+    + Headers
+
+        ```
+        Date: Fri, 29 Jan 2016 20:25:10 GMT
+        Content-Length: 26
+        Content-Type: text/plain; charset=utf-8
+        ```
+
+    + Body
 
-## wantlist
+        ```
+        Argument 'key' is required
+        ```
+
++ Request Empty Argument
+
+    #### curl
+
+        curl -i http://localhost:5001/api/v0/bitswap/unwant?arg=
+
+    + Body
+
+        ```
+        curl -i http://localhost:5001/api/v0/bitswap/unwant?arg=
+        ```
+
++ Response 500 (application/json)
+
+    + Headers
+
+        ```
+        Access-Control-Allow-Headers: X-Stream-Ouptut, X-Chunked-Output
+        Access-Control-Expose-Headers: X-Stream-Ouptut, X-Chunked-Output
+        Trailer: X-Stream-Error
+        Transfer-Encoding: chunked
+        Date: Fri, 29 Jan 2016 20:40:32 GMT
+        Transfer-Encoding: chunked
+        ```
+
+    + Attributes (Error)
+        - Message: 'incorrectly formatted key: '
+        - Code: 0
+
+    + Body
+
+        ```
+        {
+          "Message": "incorrectly formatted key: ",
```

Comment:
Inconsistent error messages. https://github.com/ipfs/http-api-spec/pull/24/files#r51342821 is the same issue, but shows a different error message.